### ISSUE TITLE
New to SSO on Jellyfin - Is there a way to keep Kodi addon login ?

### ISSUE THREAD

**Issue by LeVraiRoiDHyrule** [9p4/jellyfin-plugin-sso#70]
Hi,

I have a Jellyfin server for years and now looking to integrate it better with all my services so I'm looking to get SSO for everything. This plugin looks really cool and I'm looking forward integrating it with Authentik or Keycloak. But my problem is that I am still using the Kodi addon on some devices, even if most of my users use web.

Is there a way to get SSO and Kodi login working together ? Is there a way to still have a password that I can input in the addon ? Is there a way to have like an "app password" on Microsoft Accounts, to connect on apps that don't have a browser ?

Thanks in advance for any answer,
Have a great day

**Comment by 9p4** [comment]
Which Kodi addon? Can you provide a link?

**Comment by LeVraiRoiDHyrule** [comment]
I'm talking about the official kodi add on for jellyfin https://github.com/jellyfin/jellyfin-kodi

**Comment by 9p4** [comment]
Atm, quick connect is recommended for devices that cannot use SSO login (ie mobile apps without a web browser). This should be fixed in a later JF version (10.9 or something idk).

I'll spin up Kodi and Jellyfin and see how I can get them working together

**Comment by matthewstrasiotto** [comment]
>  Is there a way to still have a password that I can input in the addon ? 

Yep, I recommend checking out the LDAP auth plugin, as well at the setting for our plugin for setting default Auth provider.

Basically, accounts that already have passwords can continue using password Auth as their primary mechanism, if you configure the plugin to let them.

Where it gets slightly tricky is in the case where you create an account that doesn't already exist via SSO. 
The account won't be created with a password in jellyfin.
The user will manually have to configure their password.

If your IDP server either uses LDAP as a source of truth, or if it exposes an LDAP interface ( Authentik does this ), then accounts that you create via the SSO plugin can be automatically configured to use the LDAP server for password auth.

From the point of view of basically any interface that uses password Auth, LDAP backed password Auth is identical to normal Auth.

@9p4 I reckon this question/use case is common enough to deserve a proper doc/FAQ write up, WDYT? 

Also, when it comes to provisioning new accounts, I wonder if there's some way to nudge users to set their password in jellyfin?

**Comment by charredchar** [comment]
I wanted to note that this Kodi plugin [should support the Quick Connect function](https://github.com/jellyfin/jellyfin-kodi/pull/1117) in a future release. Using LDAP as a fallback solves this issue as well and is what I'm using. If you need help check issue #73 as I added some details there for using both. Useful if you already have LDAP available or want a password fallback but if you don't have that I'd keep an eye out for the plugin to update, maybe mention it there so it can be tested with SSO before release.
